File: cpp/step/dp/main_2579.cpp
#include <iostream>
#include <algorithm>

using namespace std;

int main() {
	int count;
	cin >> count;

	int dp[301] = {0,};
	int num[301] = {0,};

	cin >> num[1];
	dp[1] = num[1];
	if (count > 1) {
		cin >> num[2];
		dp[2] = dp[1] + num[2];
	}
	for (int i = 3; i <= count; ++i) {
		cin >> num[i];
		dp[i] = max(dp[i - 3] + num[i - 1] + num[i], dp[i - 2] + num[i]);
	}
	cout << dp[count] << endl;
}

File: kotlin/step/dp/main_2579.kt
import kotlin.math.max

fun main() {
    val num = readLine()!!.toInt()
    val dp = mutableListOf(0)

    var prev = readLine()!!.toInt()
    dp.add(prev)

    var now = 0
    if (num > 1) {
        now = readLine()!!.toInt()
        dp.add(prev + now)
        prev = now
    }

    for (i in 3..num){
        now = readLine()!!.toInt()
        dp.add(max(dp[i - 3] + prev + now, dp[i - 2] + now))
        prev = now
    }
    println(dp[num])
}